scalingo/resource_scalingo_database.go

```diff
@@ -0,0 +1,251 @@
+	if err != nil {
+		return diag.Errorf("wait for the addon to be provisioned: %v", err)
+	}
+
+	d.SetId(res.ID)
+
+	err = d.Set("app_id", res.App.ID)
+	if err != nil {
+		return diag.Errorf("store app id: %v", err)
+	}
+
+	err = d.Set("database_id", res.Database.ID)
+	if err != nil {
+		return diag.Errorf("store database id: %v", err)
+	}
+
+	return nil
+}
+
+func resourceDatabaseRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, _ := meta.(*scalingo.Client)
+	previewClient := scalingo.NewPreviewClient(client)
+
+	database, err := previewClient.DatabaseShow(ctx, d.Id())
+	if err != nil {
+		return diag.Errorf("get database details: %v", err)
+	}
+
+	d.SetId(database.ID)
+
+	planID, err := addonPlanID(ctx, client, database.Technology, database.Plan)
+	if err != nil {
+		return diag.Errorf("get addon plan id: %v", err)
+	}
+
+	err = SetAll(d, map[string]interface{}{
+		"name":        database.Name,
+		"technology":  database.Technology,
+		"plan":        database.Plan,
+		"plan_id":     planID,
+		"project_id":  database.ProjectID,
+		"app_id":      database.App.ID,
+		"database_id": database.Database.ID,
+	})
+	if err != nil {
+		return diag.Errorf("store database information: %v", err)
+	}
+
+	return nil
+}
+
+func resourceDatabaseUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, _ := meta.(*scalingo.Client)
+	previewClient := scalingo.NewPreviewClient(client)
+
+	database, err := previewClient.DatabaseShow(ctx, d.Id())
+	if err != nil {
+		return diag.Errorf("get addon information for %v: %v", d.Id(), err)
+	}
+
+	if d.HasChange("name") {
+		app, err := client.AppsShow(ctx, database.App.ID)
+		if err != nil {
+			return diag.Errorf("fetch database application: %v", err)
+		}
+
+		_, err = client.AppsRename(ctx, app.Name, d.Get("name").(string))
+		if err != nil {
+			return diag.Errorf("rename database app: %v", err)
+		}
+	}
+
+	if d.HasChange("project_id") {
+		_, stackID := d.GetChange("project_id")
+		_, err := client.AppsSetProject(ctx, database.App.ID, stackID.(string))
+		if err != nil {
+			return diag.Errorf("set project ID: %v", err)
+		}
+	}
+
+	return nil
+}
+
+func resourceDatabaseDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, _ := meta.(*scalingo.Client)
+
+	id := d.Get("app_id").(string)
+	name, _ := d.Get("name").(string)
+
+	err := client.AppsDestroy(ctx, id, name)
+	if err != nil {
+		return diag.Errorf("destroy app: %v", err)
+	}
+	return nil
+}
+
+func resourceDatabaseImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	client, _ := meta.(*scalingo.Client)
+	previewClient := scalingo.NewPreviewClient(client)
+
+	// d.Id() contains the database name provided by the user during import corresponding to the app name
+	// We need to find the App ID associated with this app name to retrieve the database
+
```

Comment:
To be sure to understand, here the app name is in fact the database name, right?